### PR TITLE
Initial add TableStore streaming source

### PR DESCRIPTION
**Pull request by okingniko**
Initial add TableStore streaming source

### CODE REVIEW

File: emr-tablestore/pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
         <dependency>
             <groupId>org.apache.spark</groupId>
             <artifactId>spark-core_${scala.binary.version}</artifactId>
-            <scope>provided</scope>
+             <scope>provided</scope>
```
Comment (uncleGen) [inline emr-tablestore/pom.xml]: indent

File: emr-tablestore/.scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version=2.0.0-RC5
```
Comment (uncleGen) [inline emr-tablestore/.scalafmt.conf]: This file can be deleted?

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/ChannelOffset.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.apache.spark.sql.aliyun.tablestore
```
Comment (uncleGen) [inline emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/ChannelOffset.scala]: missing license header

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreCatalog.scala [hunk 4]
```diff
@@ -35,14 +35,17 @@ object TableStoreCatalog {
...
+    )
     new TableStoreCatalog(schema)
   }
 
   def getColsPreservingOrder(jObj: JObject): Seq[(String, Map[String, String])] = {
     val jCols = jObj.obj.find(_._1 == columns).get._2.asInstanceOf[JObject]
-    jCols.obj.map { case (name, jvalue) =>
-      (name, jvalue.values.asInstanceOf[Map[String, String]])
+    jCols.obj.map {
+      case (name, jvalue) =>
```
Comment (uncleGen) [inline emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreCatalog.scala]: minor: previous code style is just ok

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSink.scala [hunk 5]
```diff
@@ -24,7 +24,9 @@ import org.apache.spark.sql.types.StructType
 class TableStoreSink(
     parameters: Map[String, String],
     userSpecifiedschema: Option[StructType]
-  )(@transient val sqlContext: SQLContext) extends Sink with Logging {
+)(@transient val sqlContext: SQLContext)
+    extends Sink
+    with Logging {
```
Comment (uncleGen) [inline emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSink.scala]: minor: keep the previous code style here and other places

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSource.scala [hunk 6]
```diff
@@ -17,23 +17,359 @@
 
 package org.apache.spark.sql.aliyun.tablestore
 
+import com.alicloud.openservices.tablestore.model._
+import com.alicloud.openservices.tablestore.model.tunnel.internal.{
+  CheckpointRequest,
+  GetCheckpointRequest
+}
```
Comment (uncleGen) [inline emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSource.scala]: minor: keep in same line

File: emr-tablestore/pom.xml (hunk 1 above)
Comment (okingniko) [reply emr-tablestore/pom.xml]: fixed

File: emr-tablestore/.scalafmt.conf (hunk 2 above)
Comment (okingniko) [reply emr-tablestore/.scalafmt.conf]: fixed

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/ChannelOffset.scala (hunk 3 above)
Comment (okingniko) [reply emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/ChannelOffset.scala]: all files add license header

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreCatalog.scala (hunk 4 above)
Comment (okingniko) [reply emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreCatalog.scala]: checkout to the previous version

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSink.scala (hunk 5 above)
Comment (okingniko) [reply emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSink.scala]: done

File: emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSource.scala (hunk 6 above)
Comment (okingniko) [reply emr-tablestore/src/main/scala/org/apache/spark/sql/aliyun/tablestore/TableStoreSource.scala]: done

**Comment by uncleGen** [comment]
LGTM，thanks

**Pull request merged**